### PR TITLE
Update docs with 1.46 features for Remote Development

### PR DESCRIPTION
**Pull request by Chuxel**
This PR includes the following updates:

1. Updates that cover the new Inspect Volume feature
2. Mention perf benefits of WSL2 filesystem, refine perf doc given inspect volume improvements (https://github.com/microsoft/vscode-docs/issues/3717)
3. Updates that cover GPG key support
4. postStartCommand, postAttachCommand
5. Switches examples of postCreateCommand from using git to other options (https://github.com/microsoft/vscode-remote-release/issues/3036#issuecomment-638527956)
6. Mention Inherit Env as a requirement for remoteEnv to work in terminals (https://github.com/microsoft/vscode-remote-release/issues/3030#issuecomment-639720678)
7. Update Alpine minimum to 3.8 (https://github.com/microsoft/vscode-remote-release/issues/3024)

### CODE REVIEW

File: docs/remote/containers.md [hunk 1]
```diff
@@ -556,48 +575,51 @@ First, start the SSH Agent in the background by running the following in a termi
...
+### Sharing GPG Keys
 
-By default, the Remote - Containers extension automatically starts the containers mentioned in the `devcontainer.json` when you open the folder. When you close VS Code, the extension automatically shuts down the containers you've connected to. You can change this behavior by adding `"shutdownAction": "none"` to `devcontainer.json`.
+If want to GPG sign your commits, you can share your local keys with your container as well. You can find out about signing using a GPG key in [GitHub's documentation](https://help.github.com/en/github/authenticating-to-github/managing-commit-signature-verification).
 
-You can also manually manage your containers using one of the following options:
+If you do not have GPG set up, on **Windows**, you can install [Gpg4win](https://www.gpg4win.org/) or on **macOS** you can install [GPG Tools](https://gpgtools.org/).
 
-### Option 1: Use the Containers Remote Explorer
+On **Linux**, you can install `gnupg2` using your system's package manger **locally**. If needed you can add  `([ -S "$HOME/.gnupg/S.gpg-agent" ] || gpg-agent --daemon)` to your `.bash_profile` or `.zprofile` (for Zsh) to make sure the agent starts.
```
Comment (chrmarti) [inline docs/remote/containers.md:598]: We start the agent from the extension. Let me know if that didn't work.

File: docs/remote/containers.md [hunk 2]
```diff
@@ -556,48 +575,51 @@ First, start the SSH Agent in the background by running the following in a termi
...
+## Managing containers
+
+By default, the Remote - Containers extension automatically starts the containers mentioned in the `devcontainer.json` when you open the folder. When you close VS Code, the extension automatically shuts down the containers you've connected to. You can change this behavior by adding `"shutdownAction": "none"` to `devcontainer.json`.
 
-1. Open a **local** terminal/PowerShell (or use a local window in VS Code).
-2. Go to the directory with your `docker-compose.yml` file.
-3. Type `docker-compose top` to see running processes.
-4. Type `docker-compose stop` to stop the containers. If you have more than one Docker Compose file, you can specify additional Docker Compose files with the `-f` argument.
-5. If you would like to delete the containers, type `docker-compose down` to both stop and delete them.
+While you can use the command line to manage your containers, you can use the Containers Explorer. You can right-click on a running container and stop it. You can also start exited containers, remove containers, and remove recent folders. From the Details view, you can forward ports and open already forwarded ports in the browser.
```
Comment (chrmarti) [inline docs/remote/containers.md:620]: Should be "Remote Explorer".

File: docs/remote/containers.md [hunk 3]
```diff
@@ -1137,16 +1163,13 @@ See [Docker Desktop for Windows tips](/docs/remote/troubleshooting.md#docker-des
 
 ### Can I use Podman instead of Docker?
 
-Podman 1.9+ is mostly compatible with Docker's CLI commands and therefore generally does work if you symlink the `podman` command to `docker` command on Linux.
+Podman 1.9+ is mostly compatible with Docker's CLI commands and therefore generally does work if you pdate the setting **Remote > Containers: Docker Path** to `podman` on Linux.
```
Comment (chrmarti) [inline docs/remote/containers.md:1166]: pdate > update

File: docs/remote/containers.md [hunk 4]
```diff
@@ -186,6 +186,8 @@ For example, follow these steps to open one of the "try" repositories in a Repos
 
 5. After the build completes, VS Code will automatically connect to the container. You can now work with the repository source code in this isolated environment as you would if you had cloned the code locally.
 
+Note that if the container fails to come up due to something like a Docker build error, you can select **Inspect Volume** in the dialog that appears to go into a "recovery container" that allows you to edit your Dockerfile or other content. Once done, use **File > Open Recent** to select the repository container and retry.
```
[the remark this answers is not in the document]
Comment (egamma) [reply docs/remote/containers.md]: >Once done, use **File > Open Recent** to select the repository container and retry.

There is now also support for 'Reopen in Container' see your feature request https://github.com/microsoft/vscode-remote-release/issues/3123

**Pull request merged**